app/src/main/java/fr/free/nrw/commons/nearby/DirectUpload.java

```diff
@@ -0,0 +1,95 @@
+package fr.free.nrw.commons.nearby;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.os.Build;
+import android.support.v4.app.Fragment;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AlertDialog;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import fr.free.nrw.commons.R;
+import fr.free.nrw.commons.contributions.ContributionController;
+
+import static android.Manifest.permission.READ_EXTERNAL_STORAGE;
+import static android.Manifest.permission.WRITE_EXTERNAL_STORAGE;
+import static android.content.pm.PackageManager.PERMISSION_GRANTED;
+
+class DirectUpload {
+
+    private String title;
+    private String desc;
+    private ContributionController controller;
+    private Fragment fragment;
+
+    @Inject @Named("prefs") SharedPreferences prefs;
+    @Inject @Named("default_preferences") SharedPreferences defaultPrefs;
+
```

Comment:
There are 2 places where instances of this class are created manually and neither of them bring Dagger into the mix (lines 402 and 410 of the nearby map fragment).  Since Dagger wasn't called, these members wont have been initialized.  Since the Fragment *was* injected, these dependencies should be passed into the constructor (and `@Inject` annotations removed) - all should be well and NPE's averted.

Reply:
It seems that NearbyMapFragment was not injected, only NearbyListFragment. I guess I can inject NearbyMapFragment anyway and move these lines there?